data-prepper-plugins/aggregate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/aggregate/AggregateAction.java

```diff
@@ -25,7 +26,7 @@ public interface AggregateAction {
      * be processed immediately, or is empty if the Event should be removed from processing
      * @since 1.3
      */
-    default AggregateActionResponse handleEvent(final Event event, final AggregateActionInput aggregateActionInput) {
+    default AggregateActionResponse handleEvent(final Event event, final AggregateActionInput aggregateActionInput, final Map<String, Object> identificationKeysMap) {
```

Comment:
Can we pass the identificationKeys as part of the AggregateActionInput? I would like to keep the signature of the AggregateAction functions to be implemented as simple as possible

Reply:
keeping identificationKeys as part of `AggregateActionInput` means creating AggregateGroup with identification keys. I thought that's more involved than changing the `handleEvent()` signature. I had my initial implementation that way but it requires more changes because it needs to go through AggregateGroupManager to AggregateGroup. That looked more forced and not natural. But I am OK to do that if that makes more sense structually

Reply:
I see what you mean, but I think it's fine to pass the `IdentificationHash` to the AggregateGroup. The aggregateGroup constructor can just take the hash as input here https://github.com/opensearch-project/data-prepper/blob/b381189e04079c6e23ac98a7f32741351862c347/data-prepper-plugins/aggregate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/aggregate/AggregateGroupManager.java#L25. I would still prefer this just for the sake of keeping the AggregateAction interface cleaner.